cmd/rook/ceph/operator.go

```diff
@@ -19,6 +19,7 @@ package ceph
 import (
 	"github.com/pkg/errors"
 	"github.com/rook/rook/cmd/rook/rook"
+	"github.com/rook/rook/cmd/webhook-server"
```

Comment:
Some suggestions on the code structure:
1. We shouldn't need any changes under the `cmd/rook/ceph/` package.
   - If we decide to run the admission controller as a separate daemon instead of part of the operator, there would only be small changes in this package for a new command line option (`rook ceph admission-controller`). The handler for this new option would call the code under `pkg` in the next bullet.
1. The code under `cmd/rook/webhook-server` should be moved under `pkg/operator/ceph/admission`.
1. All of the changes in this file likely should also move under `pkg/operator/ceph/admission`

Reply:
Per our discussion, there are two phases of implementation to consider:
1. Changes to the operator that will launch the admission controller 
   - Check if the secret exists that signals the admission controller is expected to run. 
   - If the secret exists:
     - Create the K8s service that directs the traffic to the admission controller daemon 
     - Create the admission controller deployment spec and start it. 
1. New `rook admission-controller ceph` command that implements running the admission controller daemon

Note that all creation of k8s resources should be done in phase one. Phase two should not need to create resources, it just starts listening for the webhook events.